mtools/mlaunch/mlaunch.py

```diff
@@ -1696,7 +1696,12 @@ def _initiate_replset(self, port, name, maxwait=30):
         con['admin'].command({'ping': 1})
 
     def _add_user(self, port, name, password, database, roles):
-        con = self.client('localhost:%i' % port)
+        con = self.client('localhost:%i' % port, serverSelectionTimeoutMS=10000)
+        ismaster = con['admin'].command('isMaster')
+        set_name = ismaster.get('setName')
+        if set_name:
+            con.close()
+            con = self.client('localhost:%i'%port, replicaSet=set_name, serverSelectionTimeoutMS=10000)
         v = con['admin'].command('isMaster').get('maxWireVersion', 0)
```

Comment:
We could optimize this code path to only run isMaster once by re-using the previous `ismaster` response for the maxWireVersion check. I'll leave that up to you.

Reply:
Thanks for the assist, @ShaneHarvey!

@p-mongo can you make the last change suggested?

Reply:
What exactly is the change that is desired?

Reply:
```
...
if set_name:
    con.close()
    con = ....
    ismaster = con['admin'].command('isMaster')
v = ismaster.get('maxWireVersion', 0)
```